File: rust/tests/trace_streams_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_trace_streams_stream_openinference_spans_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .trace_streams
        .stream_openinference_spans(
            &TraceStreamRequest {
                format: TraceStreamRequestFormat::OpeninferenceV1,
                metadata: TraceMetadata {
                    account_id: "account_id".to_string(),
                    project_id: "project_id".to_string(),
                    source: "source".to_string(),
                    ..Default::default()
                },
                dataset: None,
                resource_spans: None,
                spans: None,
                trajectories: None,
                trajectory: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/trace-streams", None, 1)
        .await
        .unwrap();
}

File: ruby/test/wire/trace_streams_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class TraceStreamsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_trace_streams_stream_openinference_spans_with_wiremock
    test_id = "trace_streams.stream_openinference_spans.0"

    @client.trace_streams.stream_openinference_spans(
      format: "openinference.v1",
      metadata: {
        account_id: "account_id",
        project_id: "project_id",
        source: "source"
      },
      request_options: {
        additional_headers: {
          "X-Test-Id" => "trace_streams.stream_openinference_spans.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/trace-streams",
      query_params: nil,
      expected: 1
    )
  end
end
